scripts/ubuntu-bartender/qemu-provider

```diff
@@ -0,0 +1,141 @@
+function build-provider-assert-ready {
+  if ! command -v qemu-system-${QEMU_ARCH} &>/dev/null
+  then
+    echo "error: qemu-system-${QEMU_ARCH} was not found in PATH" >&2
+    exit 255
+  fi
+
+  if ! command -v cloud-localds &>/dev/null; then
+    echo "error: cloud-image-utils not found. You can install it with:"
+    echo "sudo apt install -y cloud-image-utils"
+    exit 255
+  fi
+
+  if [ -n "$QEMU_IMAGE" ]; then
+    if [ ! -f "$QEMU_IMAGE" ]; then
+      echo "error: specified image not found at $QEMU_IMAGE"
+      exit 255
+    fi
+    virtual_size=$(qemu-img info --output=json "$QEMU_IMAGE" | jq '."virtual-size"')
+    if [ "$virtual_size" -lt 50000000000 ]; then
+      echo "error: qemu disk is too small, it should be at least 50GB large"
```

Comment:
So 2 things here:
 1. The largest image is Azure: 30GB
 2. This is a "sparse" disk, it doesn't actually takes 50GB on your disk:
  ```
➜  /tmp ll bionic-server-cloudimg-amd64.img
-rw-rw-r-- 1 gauthier gauthier 354M Jun 23 04:28 bionic-server-cloudimg-amd64.img
➜  /tmp qemu-img resize /tmp/bionic-server-cloudimg-amd64.img +50G
Image resized.
➜  /tmp ll bionic-server-cloudimg-amd64.img
-rw-rw-r-- 1 gauthier gauthier 354M Jul  9 14:32 bionic-server-cloudimg-amd64.img
➜  /tmp qemu-img info bionic-server-cloudimg-amd64.img
image: bionic-server-cloudimg-amd64.img
file format: qcow2
virtual size: 52.2 GiB (56048484352 bytes)
disk size: 354 MiB
cluster_size: 65536
Format specific information:
    compat: 0.10
    compression type: zlib
    refcount bits: 16
  ```

Reply:
@j5awry I think 50GBcame from me with the AWS provider (or GCE).  It was arbitrary but based on running out of space when building all image targets on releases that have a great number of them, disks of smaller size kept running out of space and breaking my builds.  Finally I moved to 50GB and didn't have the issues any longer but I can't tell you what the high-water mark is for usage when all targets are selected.  As @gjolly said, it's a sparse disk so you won't see it consume all that space most days.

*But* @gjolly  it has the potential to consume 50GB and if it does then your host (laptop) could run out of disk space (small, full SSDs?) and that might be more annoying for this local qemu provider that the build falling over.  50GB was a good, safe bet when launching an ephemeral VM in a cloud where the size is guaranteed and cheaper to allocate the excess than pay your salary to restart a build.  For the qemu-provider you might want to tune the default lower.

Reply:
+1 on tuning default a bit lower, and possible to some fun mathematical variant (like we did for multipass ram size). something like (pseudo-coded)

```
if < 25gb free space:
    echo "Low space detected on drive. please consider using a cloud provider"
    exit 1
if < 60gb free space:
    echo "less than optimal space found. Be aware that on large builds, this may reach capacity."
    size = 50% free space
else:
    size = 30gb
echo "setting sparse disk to ${size}"
```
that'd mean the absolute smallest would be around 12.5gb, which should build a single image comfortably, and maybe build one of the larger aggregates. Worth testing out that logic and gut feeling on sizes.